components/core/tests/test-encoding_methods.cpp

```diff
@@ -226,6 +226,20 @@ TEMPLATE_TEST_CASE("Encoding floats", "[ffi][encode-float]", eight_byte_encoded_
     decoded_value = decode_float_var(encoded_var);
     REQUIRE(decoded_value == value);
 
+    // Test out of range
+    if constexpr (std::is_same_v<TestType, four_byte_encoded_variable_t>) {
+        four_byte_encoded_variable_t encoded_var;
+        auto float_out_of_range = GENERATE(
```

Comment:
* We can set the type to `string` and use literals in the GENERATE expression to avoid repeating the type.
* Can we add all the below tests to a GENERATE expression?